File: models/healthmetric.py
# models/healthmetric.py
from sqlalchemy import Column, Integer, String, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from .base import Base

class HealthMetric(Base):
    __tablename__ = "healthmetric"  # matches HealthMetric table (lowercased)

    metric_id = Column(Integer, primary_key=True, autoincrement=True)
    member_id = Column(Integer, ForeignKey("member.member_id", ondelete="SET NULL"))
    metric_type = Column(String, nullable=False)
    value = Column(Integer)
    timestamp = Column(DateTime)

    member = relationship("Member", back_populates="metrics")

File: health_club_management/src/main/resources/healthMetric.java
package healthClubManagement;

import jakarta.persistence.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "HealthMetric")
public class HealthMetric {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "metric_id")
    private Long metricId;

    @ManyToOne
    @JoinColumn(name = "member_id")
    private Member member;

    @Column(name = "metric_type", nullable = false)
    private String metricType;

    @Column(name = "value")
    private Integer value;

    @Column(name = "timestamp")
    private LocalDateTime timestamp;

    // Getters and setters

    public Long getMetricId() {
        return metricId;
    }

    public void setMetricId(Long metricId) {
        this.metricId = metricId;
    }

    public Member getMember() {
        return member;
    }

    public void setMember(Member member) {
        this.member = member;
    }

    public String getMetricType() {
        return metricType;
    }

    public void setMetricType(String metricType) {
        this.metricType = metricType;
    }

    public Integer getValue() {
        return value;
    }

    public void setValue(Integer value) {
        this.value = value;
    }

    public LocalDateTime getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(LocalDateTime timestamp) {
        this.timestamp = timestamp;
    }

    @Override
    public String toString() {
        return "HealthMetric{" +
                "metricId=" + metricId +
                ", member=" + (member != null ? member.getMemberId() : null) +
                ", metricType='" + metricType + '\'' +
                ", value=" + value +
                ", timestamp=" + timestamp +
                '}';
    }
}
